lib/rust/ensogl/component/list-view/src/entry/list.rs

```diff
@@ -58,12 +58,13 @@ impl IdAtYPosition {
 ///
 /// Not all entries are displayed at once, only those visible.
 #[derive(Clone, CloneRef, Debug)]
-pub struct List<E: CloneRef> {
+pub struct List<E: Entry> {
```

Comment:
Note: this change was done to allow the use of `E::Params` below in the type of the `entry_params` field of the struct. This was consulted with @farmaazon.

Reply:
I really don't like it. We should never use bounds on struct parameters (the previous code was using them and I'm wondering why - there is no docs with justification why it is needed, and browsing the code I dont see any reason why this struct could not be defined without bounds). The only justification are bounds required by Drop impl or because a struct uses a not our library which used bounds in wrong way.

Because of this, these bounds are leaking everywhere. For example, later you have:

```
impl<E: Entry> display::Object for List<E> {
     fn display_object(&self) -> &display::object::Instance {
         &self.display_object
     }
}
```

And obviously, in order to access the display object, you dont need `E` to be `Entry`.